### PR TITLE
fix: formatting of deadline in permit signature page

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Improve formatting of deadline field in permit signature type.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2642

## **Manual testing steps**

1. Go to test DAPP
2. Submit permit signature type
3. Check formatting of deadline field

## **Screenshots/Recordings**
Screenshot 2024-06-14 at 5 56 34 PM

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/confirm/info/personal-sign/siwe-sign/siwe-sign.tsx [hunk 1]
```diff
@@ -64,7 +65,9 @@ const SIWESignInfo: React.FC = () => {
         <ConfirmInfoRowText text={nonce} />
       </ConfirmInfoRow>
       <ConfirmInfoRow label={t('siweIssued')}>
-        <ConfirmInfoRowText text={formatDate(issuedAt)} />
+        <ConfirmInfoRowDate
+          date={DateTime.fromISO(issuedAt).toJSDate().getTime()}
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/personal-sign/siwe-sign/siwe-sign.tsx:69]: The functions are similar but bit different - in parameters, in timezone.

File: ui/pages/confirmations/components/confirm/row/dataTree.tsx [hunk 2]
```diff
@@ -20,32 +21,50 @@ export type TreeData = {
...
+  label: string;
+  isPermit: boolean;
+  type: string;
+  value: ValueType;
+}) => {
   if (typeof value === 'object' && value !== null) {
-    return <DataTree data={value} />;
+    return <DataTree data={value} isPermit={isPermit} />;
+  }
+  if (isPermit && label === 'deadline') {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/row/dataTree.tsx:66]: This component is still very tied to typed sign data and I do not see we will be able to use it anywhere else. Thus I would avoid incresing complexity to make it generic.

File: ui/pages/confirmations/components/confirm/info/personal-sign/siwe-sign/siwe-sign.tsx [hunk 3]
```diff
@@ -9,9 +10,9 @@ import { SignatureRequestType } from '../../../../../types/confirm';
 import {
   ConfirmInfoRow,
   ConfirmInfoRowAddress,
+  ConfirmInfoRowDate,
   ConfirmInfoRowText,
 } from '../../../../../../../components/app/confirm/info/row';
-import { formatDate } from '../../../../../utils/date';
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/info/personal-sign/siwe-sign/siwe-sign.tsx:14]: Its used at a couple of other places.

File: ui/components/app/confirm/info/row/date.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import { ConfirmInfoRow } from './row';
+import { ConfirmInfoRowDate } from './date';
+
+const ConfirmInfoRowDateStory = {
+  title: 'Components/App/Confirm/InfoRowDate',
+  component: ConfirmInfoRowDate,
+
+  decorators: [
+    (story) => <ConfirmInfoRow label="Message">{story()}</ConfirmInfoRow>,
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/components/app/confirm/info/row/date.stories.tsx]: Fixed

File: ui/helpers/utils/util.js [hunk 5]
```diff
@@ -43,6 +43,17 @@ export function formatDate(date, format = "M/d/y 'at' T") {
   return DateTime.fromMillis(date).toFormat(format);
 }
 
+export const formatUTCDate = (dateInMillis) => {
+  if (!dateInMillis) {
+    return dateInMillis;
+  }
+
+  return DateTime.fromMillis(dateInMillis)
+    .setLocale('en')
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/helpers/utils/util.js]: Looks like this was redundant, I removed it

File: ui/pages/confirmations/components/confirm/row/dataTree.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewwalsh0) [reply ui/pages/confirmations/components/confirm/row/dataTree.tsx:66]: But would it be fair to say we're already increasing the complexity with this additional coupling and logic and knowledge of specific fields?

It's the same level of additional complexity, but it keeps the knowledge of the `deadline` field and permit in one place so it's the concept of a permit remains encapsulated and we don't introduce any domain into these great row components?

[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/row/dataTree.tsx:66]: We can write separate may be utility functions to check fields - we have access to current confirmation and data here, but declaring these in parent component and then passing 2 levels down will not be great idea I think.

I will prefer to get into re-factoring this component only if its use gets more complicated.

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**